sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,196 @@
+	protected static final String SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE = "sectionsToolPropsPathLessSecure";
+	protected static final String PROPERTY_CANVAS_ADMIN = "canvas.admin.token";
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_USE_TEST_URL = "use.test.url";
+	protected static final String PROPERTY_LDAP_SERVER_URL = "ldap.server.url";
+	private static final String PROPERTY_AUTH_GROUP = "mcomm.group";
+	private static final String TEST_USER = "testUser";
+	private String providerURL = null;
+	private String mcommunityGroup = null;
+	private boolean isTestUrlEnabled=false;
+	protected static Properties appExtSecurePropertiesFile=null;
+	protected static Properties appExtLessSecurePropertiesFile=null;
+	private static final String FALSE = "false";
+	
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getExternalAppProperties();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+			useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("destroy: Called");
+	}
+	
+	protected void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePathSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_SECURE);
+		String propertiesFilePathLessSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE);
+		if (!isEmpty(propertiesFilePathSecure)) {
+			appExtSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathSecure);
+			if(appExtSecurePropertiesFile!=null) {
+				isTestUrlEnabled = Boolean.parseBoolean(appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_USE_TEST_URL,FALSE));
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathSecure.properties) is not provided");
+		}
+		
+		if (!isEmpty(propertiesFilePathLessSecure)) {
+			appExtLessSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathLessSecure);
+			if(appExtLessSecurePropertiesFile!=null) {
+				providerURL=appExtLessSecurePropertiesFile.getProperty(PROPERTY_LDAP_SERVER_URL);
+				mcommunityGroup=appExtLessSecurePropertiesFile.getProperty(PROPERTY_AUTH_GROUP);
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsLessSecure.properties for SectionsTool");
```

Comment:
Why is this logged as a warning, while the absence of the the environment variable logged as an error? Both seem to result in a fatal condition where the app can't run, so I'd log this as an error.